scripts/spotify-app.js

```diff
@@ -4,6 +4,65 @@
 $(document).ready(function() {
     console.log('ready');
 
+    // Authorization info
+    var CLIENT_ID = "d96dcdaf6e674b46a1bf42d5f6367889";
+    var AUTH_ENDPOINT = "https://accounts.spotify.com/authorize";
+    var REDIRECT_URI = "http://alexpersian.com/html/spotify-app.html";
+
+    // Implicit Grant auth process
+    // Referencing Spotify's example
+    var stateKey = 'auth_state_key';
+
+    // Parse out the parameters from the URL hash
+    function getHashParams() {
+        var hashParams = {};
+        var e, r = /([^&;=]+)=?([^&;]*)/g,
+            q = window.location.hash.substring(1);
+        while (e = r.exec(q)) {
+            hashParams[e[1]] = decodeURIComponent(e[2]);
+        }
+        return hashParams;
+    };
+
+    // Random string used for state value
+    function generateRandomString(length) {
```

Comment:
I think this is cleaner:

Math.random().toString(36)

Get a random number between 0,1, convert to a string but in base 36, and then take a substring of that.